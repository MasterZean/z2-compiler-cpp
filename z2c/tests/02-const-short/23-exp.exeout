var a: Short = Short{0}
var b: Short = Short{0}
var c: Short = Short{-1}
var d: Short = Short{1}
var e: Short = Short{-100}
var f: Short = Short{100}
var g: Short = Short{-127}
var h: Short = Short{127}
var i: Short = Short{-128}
var j: Short = Short{128}
var k: Short = Short{-200}
var l: Short = Short{200}
var m: Short = Short{-32767}
var n: Short = Short{32767}
var o: Short = Short{-32768}
