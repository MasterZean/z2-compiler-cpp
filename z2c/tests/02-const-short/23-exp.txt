val a: Short = -0s32
val b: Short = 0s32
val c: Short = -1s32
val d: Short = 1s32
val e: Short = -100s32
val f: Short = 100s32
val g: Short = -127s32
val h: Short = 127s32
val i: Short = -128s32
val j: Short = 128s32
val k: Short = -200s32
val l: Short = 200s32
val m: Short = -32767s32
val n: Short = 32767s32
val o: Short = -32768s32
val p: Short = 32768s32
val q: Short = -40000s32
val r: Short = 40000s32

