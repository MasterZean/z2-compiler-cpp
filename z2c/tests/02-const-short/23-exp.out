int16 a = (int16)(0);
int16 b = (int16)(0);
int16 c = (int16)(-1);
int16 d = (int16)(1);
int16 e = (int16)(-100);
int16 f = (int16)(100);
int16 g = (int16)(-127);
int16 h = (int16)(127);
int16 i = (int16)(-128);
int16 j = 128;
int16 k = -200;
int16 l = 200;
int16 m = -32767;
int16 n = 32767;
int16 o = -32768;
DummyClass(16, 14): error:
	can't assign 'Int' instance to 'Short' instance
DummyClass(17, 14): error:
	can't assign 'Int' instance to 'Short' instance
DummyClass(18, 14): error:
	can't assign 'Int' instance to 'Short' instance
